<a href="https://colab.research.google.com/github/omjiverma/Exploratory-Data-Analysis-Portfolio/blob/main/EDA_Exports_and_Imports_of_India(1997_2022).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Analysis of India's import and export
Even before independence, the Government of India maintained semi-autonomous 
diplomatic relations. It had colonies (such as the Aden Settlement), who sent and received full missions, and was a founder member of both the League of Nations and the United Nations. After India gained independence from the United Kingdom in 1947, it soon joined the Commonwealth of Nations and strongly supported independence movements in other colonies, like the Indonesian National Revolution. The partition and various territorial disputes, particularly that over Kashmir, would strain its relations with Pakistan for years to come. During the Cold War, India adopted a foreign policy of non-alignment policy itself with any major power bloc. However, India developed close ties with the Soviet Union and received extensive military support from it.


---
<img src='https://images.unsplash.com/photo-1578575437130-527eed3abbec?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2070&q=80'>
---


India exports approximately 7500 commodities to about 190 countries, and imports around 6000 commodities from 140 countries.[11] India exported US$318.2 billion and imported $462.9 billion worth of commodities in 2014.


---


Trade in services
India was the eighth largest exporter of commercial services in the world in 2016, accounting for 3.4% of global trade in services. India recorded a 5.7% growth in services trade in 2016.


---

In this EDA, I will explore the india import export data from 1997 - 2022 dataset through data visualizations and graphs using plotly and pandas.

---


**Note**:DATA IS IN US DOLLARS(IN MILLIONS)

---




##Install Package and Import

In [1]:
!pip install opendatasets chart_studio -q

     |████████████████████████████████| 64 kB 2.4 MB/s 


### Import required libraries

In [2]:
import opendatasets as od
import pandas as pd
import plotly.express as px
import numpy as np

### Download Dataset from kaggle

In [10]:
URL='https://www.kaggle.com/ramjasmaurya/exports-and-imports-of-india19972022'
od.download_kaggle_dataset(URL,'./data')

Skipping, found downloaded files in "./data/exports-and-imports-of-india19972022" (use force=True to force download)


### Loading the Dataset
We will pandas read_csv function to read data csv file and load it our notebook

In [11]:
dataset=pd.read_csv('./data/exports-and-imports-of-india19972022/exports and imports of india(1997-2022) - exports and imports.csv')

In [12]:
dataset.head()

Country Export Import Total Trade Trade Balance  Year(start)  Year(end)
0  AFGHANISTAN  21.25   10.7       31.95         10.55           97         98
1  AFGHANISTAN  12.81  28.14       40.95        -15.33           98         99
2  AFGHANISTAN   33.2  21.06       54.26         12.15           99       2000
3  AFGHANISTAN  25.86  26.59       52.45         -0.73         2000       2001
4  AFGHANISTAN  24.37  17.52       41.89          6.85         2001       2002

After a quick view at the dataset, we can say that our dataset has 7 columns.

# Data Preparation and Cleaning

In [13]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5767 entries, 0 to 5766
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Country        5767 non-null   object
 1   Export         5759 non-null   object
 2   Import         5235 non-null   object
 3   Total Trade    5201 non-null   object
 4   Trade Balance  5201 non-null   object
 5   Year(start)    5767 non-null   int64 
 6   Year(end)      5767 non-null   int64 
dtypes: int64(2), object(5)
memory usage: 315.5+ KB


There are 5767 entries and 7 columns to work with for EDA. There are some  columns that contain null values. But Only two coulmn has numerical data, we see above there are 6 numerical column  [Export,Import,Total Trade, Trade Balance,Year(start), Year(end)]


So we first convert [Export,Import,Total Trade, Trade Balance] to numerical type data

## Fill nan with 0

In [14]:
dataset = dataset.fillna(0)

## Coverting some columns to numerical type

To covert columns to numrical type we will use pd.to_numeric method

### First Removing special character like comma from columns converting to numerical

In [15]:
dataset[['Export','Import','Total Trade', 'Trade Balance']]=dataset[['Export','Import','Total Trade', 'Trade Balance']].replace('\,','',regex=True).astype(float)

In [16]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5767 entries, 0 to 5766
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Country        5767 non-null   object 
 1   Export         5767 non-null   float64
 2   Import         5767 non-null   float64
 3   Total Trade    5767 non-null   float64
 4   Trade Balance  5767 non-null   float64
 5   Year(start)    5767 non-null   int64  
 6   Year(end)      5767 non-null   int64  
dtypes: float64(4), int64(2), object(1)
memory usage: 315.5+ KB


We have converted required column to numerical type data

# Exploratory Data Analysis

### How many contries we export to and import from ?

In [17]:
len(dataset['Country'].unique())

250

### What is the trend of import and export from 1997-2022 ?

In [18]:
df_year = (dataset.groupby('Year(start)')['Import','Export','Trade Balance'].sum()).reset_index()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.
  """Entry point for launching an IPython kernel.


In [19]:
df_year

Year(start)     Import     Export  Trade Balance
0            97   41487.38   34785.07       -6830.45
1            98   42404.29   33218.79       -9344.27
2            99   49740.66   36822.55      -12979.44
3          2000   50553.54   44560.24       -6073.61
4          2001   51418.06   43826.71       -7664.20
5          2002   61412.12   52719.44       -8783.87
6          2003   78149.15   63842.53      -14415.70
7          2004  107134.53   83535.95      -23622.67
8          2005  129707.89  103090.53      -26660.30
9          2006  185735.30  126414.11      -59329.93
10         2007  251654.02  163132.18      -88556.63
11         2008  303696.33  185295.32     -118411.22
12         2009  288375.89  178751.47     -109650.21
13         2010  369769.14  249815.60     -119956.47
14         2011  489319.55  305963.94     -183368.66
15         2012  490736.67  300400.59     -190486.42
16         2013  450197.66  314405.32     -135801.06
17         2014  447964.44  310338.41     -137627.65
18         2015  381008.47  262291.07     -119919.34
19         2016  384356.36  275852.45     -109884.02
20         2017  465580.26  303526.16     -162102.78
21         2018  514078.34  330078.11     -184008.16
22         2019  474717.71  313361.21     -161370.59
23         2020  394435.93  291808.51     -102629.19
24         2021  440923.49  305026.06     -135927.83

Replacing year 97,98 and 99 with 1997,1998,1999

In [20]:
df_year['Year(start)'][0:3] = [1997,1998,1999]


/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [21]:
fig = px.line(df_year, x='Year(start)', y=['Import','Export'],labels={"Year(start)": "Year","value":"Amount US$ Million"}, 
              title='India\'s Import and export in US$ Million',template='xgridoff')
fig.update_layout(margin=dict(r=20),)

fig.show()

**Infrerence from the plot**
* Export remain less as compared to Import
* Import decreases from 2012 to 2016 and the decrease is very sharp in (2014-2016)
* Import has vey sharp decrease in year 2019-2020 due to covid
* Export start again after 2015 and in 2022 is also shows positive slope


## Which are the top 10 countries from which india imported in year 2021?

In [22]:
df_2021 =dataset[dataset['Year(start)'] == 2021]

In [23]:
fig = px.bar((df_2021.sort_values("Import",ascending=False))[:10], x='Import', y='Country',orientation='h',template='xgridoff',title='Top 10 Import Countries 2021',color='Country',
             labels={"Import":"Amount in US $ Million"},text_auto=True)
fig.update_layout(margin=dict(l=115))
fig.show()

* China is top Exporter country to india (67.62K Million US $)

* UAE is second top country Exporter country to india (32.71K Million US $)

* USA is second top country Exporter country to india (22.64K Million US $)

## Which are the top 10 countries to which india Exported in year 2021?

In [24]:
fig = px.bar((df_2021.sort_values("Export",ascending=False))[:10], x='Export', y='Country',orientation='h',template='xgridoff',title='Top 10 Export Countries 2021',color='Country',
             labels={"Export":"Amount in US $ Million"},text_auto=True)
fig.update_layout(margin=dict(l=115))
fig.show()

* USA is biggest importer to india impoted goods of around 55.78k US $

* UAE is Second biggest importer to india impoted goods of around 20.05k US $

* China is biggest importer to india impoted goods of around 17.12k US $

## Which are the common top 10 Countries in Import and Export ?

* USA
* UAE
* China
* Hong Cong
* Singapore


## What is distribution of Import and export over the year and find the year max and min import export value?

In [40]:
fig = px.histogram(df_year, x="Year(start)", y=["Export","Import"],nbins=25,title='Distribution of Import & Export 1997-2022')
fig.update_layout(margin=dict(l=50,r=50))
fig.show()

* Import is maximum in year 2018
* Export is maximum in year 2018

* Import minimum in year 1997
* Export minimum in year 1997


## Compare import and export over time

In [44]:
fig = px.scatter(dataset[dataset['Import'] > 0],  y="Import",x='Year(start)',
	         size="Export", color="Country",
                 hover_name="Country", size_max=60,range_x=(1997,2022))
fig.show()


* Import and export from US and UAE is continously increase

## Find the countries to which india export but do not import anything in 2021?

In [73]:
imp_tr = dataset['Import'] == 0
exp_tr = dataset['Export'] > 0
dataset[imp_tr & exp_tr][dataset[imp_tr & exp_tr]['Year(start)'] == 2021].reset_index()

index       Country  Export  Import  Total Trade  Trade Balance  \
0      99   AMERI SAMOA    0.62     0.0         0.62           0.62   
1     124       ANDORRA    0.05     0.0         0.00           0.00   
2     577       BERMUDA    4.17     0.0         4.18           4.17   
3     852    C AFRI REP   13.02     0.0         0.00           0.00   
4    1246       COOK IS    0.04     0.0         0.00           0.00   
5    1711     FAROE IS.    0.20     0.0         0.00           0.00   
6    2050     GREENLAND    1.40     0.0         0.00           0.00   
7    2871       LESOTHO    9.30     0.0         0.00           0.00   
8    3342    MICRONESIA    0.51     0.0         0.00           0.00   
9    3445    MONTSERRAT    0.08     0.0         0.00           0.00   
10   3945         PALAU    0.09     0.0         0.00           0.00   
11   4397      SAO TOME    1.79     0.0         0.00           0.00   
12   4831    ST VINCENT    0.67     0.0         0.67           0.66   
13   5112   TIMOR LESTE   18.16     0.0        18.16          18.16   
14   5183         TONGA    1.21     0.0         0.00           0.00   
15   5333        TUVALU    0.10     0.0         0.00           0.00   
16   5585   VANUATU REP    2.60     0.0         0.00           0.00   
17   5593  VATICAN CITY    0.03     0.0         0.00           0.00   

    Year(start)  Year(end)  
0          2021       2022  
1          2021       2022  
2          2021       2022  
3          2021       2022  
4          2021       2022  
5          2021       2022  
6          2021       2022  
7          2021       2022  
8          2021       2022  
9          2021       2022  
10         2021       2022  
11         2021       2022  
12         2021       2022  
13         2021       2022  
14         2021       2022  
15         2021       2022  
16         2021       2022  
17         2021       2022

There 18 countries to which india export but not import

## Find the countries from which india import but do not export anything in 2021?

In [75]:
imp_tr = dataset['Import'] > 0
exp_tr = dataset['Export'] == 0
dataset[imp_tr & exp_tr][dataset[imp_tr & exp_tr]['Year(start)'] == 2021].reset_index()

Empty DataFrame
Columns: [Country, Export, Import, Total Trade, Trade Balance, Year(start), Year(end)]
Index: []

No countries found from which india import but do not export 2021

## Find the countries from which india imported but do not exported anything year 1997 - 2022?

In [81]:
imp_tr = dataset['Import'] > 0
exp_tr = dataset['Export'] == 0
dataset[imp_tr & exp_tr].reset_index(drop=True)


Country  Export  Import  Total Trade  Trade Balance  Year(start)  \
0         ANTARTICA     0.0    4.84         4.84          -4.83         2014   
1         ANTARTICA     0.0    1.11         1.11          -1.10         2015   
2         ANTARTICA     0.0    0.03         0.04          -0.03         2018   
3     CHRISTMAS IS.     0.0    0.01         0.02          -0.01         2019   
4          COCOS IS     0.0    0.05         0.05          -0.04         2008   
5           COOK IS     0.0    0.03         0.03          -0.02         2006   
6       FALKLAND IS     0.0    1.44         1.44          -1.44         2012   
7       FALKLAND IS     0.0    1.75         1.75          -1.74         2015   
8         FAROE IS.     0.0    0.08         0.08          -0.08           98   
9       FR S ANT TR     0.0    0.01         0.01          -0.01         2010   
10        GREENLAND     0.0    0.05         0.05          -0.05         2006   
11  HEARD MACDONALD     0.0    0.03         0.03          -0.03         2008   
12  HEARD MACDONALD     0.0    0.16         0.16          -0.16         2012   
13         NAURU RP     0.0   21.03        21.03         -21.03         2012   
14          NIUE IS     0.0    0.06         0.07          -0.06         2009   
15       SAN MARINO     0.0    0.64         0.64          -0.64         2013   
16        ST PIERRE     0.0    0.02         0.02          -0.02         2008   
17       TOKELAU IS     0.0    0.01         0.02          -0.01         2009   
18       TOKELAU IS     0.0    0.02         0.02          -0.02         2018   
19       TOKELAU IS     0.0    0.08         0.08          -0.08         2019   
20      WALLIS F IS     0.0    0.10         0.10          -0.10         2013   

    Year(end)  
0        2015  
1        2016  
2        2019  
3        2020  
4        2009  
5        2007  
6        2013  
7        2016  
8          99  
9        2011  
10       2007  
11       2009  
12       2013  
13       2013  
14       2010  
15       2014  
16       2009  
17       2010  
18       2019  
19       2020  
20       2014

There are 21 countries found from year 1997 to 2022 from which india imported but do not exported 

## What is the total trade value we imported and exported in year 2021?

In [119]:
imp_exp = (dataset.groupby('Year(start)')['Import','Export','Total Trade'].sum()).reset_index();
pd.DataFrame(imp_exp.loc[24])

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning:

Indexing with multiple keys (implicitly converted to a tuple of keys) will be deprecated, use a list instead.



24
Year(start)    2021.00
Import       440923.49
Export       305026.06
Total Trade  745919.19